src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -3821,6 +3820,279 @@ internal new NamedTypeSymbol CreateNativeIntegerTypeSymbol(bool signed)
             return this.AnonymousTypeManager.ConstructAnonymousTypeSymbol(descriptor).GetPublicSymbol();
         }
 
+        protected override IMethodSymbol CommonCreateBuiltinOperator(
+            string name,
+            ITypeSymbol returnType,
+            ITypeSymbol leftType,
+            ITypeSymbol rightType)
+        {
+            var csharpReturnType = returnType.EnsureCSharpSymbolOrNull(nameof(returnType));
+            var csharpLeftType = leftType.EnsureCSharpSymbolOrNull(nameof(leftType));
+            var csharpRightType = rightType.EnsureCSharpSymbolOrNull(nameof(rightType));
+
+            // caller already checked all of these were not null.
+            Debug.Assert(csharpReturnType is not null);
+            Debug.Assert(csharpLeftType is not null);
+            Debug.Assert(csharpRightType is not null);
+
+            var syntaxKind = SyntaxFacts.GetOperatorKind(name);
+            if (syntaxKind == SyntaxKind.None)
+                throw new ArgumentException($"Illegal operator name '{name}'", nameof(name));
+
+            var binaryOperatorName = OperatorFacts.BinaryOperatorNameFromSyntaxKindIfAny(syntaxKind, SyntaxFacts.IsCheckedOperator(name));
+            if (binaryOperatorName != name)
+                throw new ArgumentException($"'{name}' was not a valid binary operator name", nameof(name));
+
+            // Lang specific checks to ensure this is an acceptable operator.
+            checkOperatorKinds();
+
+            return new SynthesizedIntrinsicOperatorSymbol(csharpLeftType, name, csharpRightType, csharpReturnType).GetPublicSymbol();
+
+            static bool isDynamicOrError(TypeSymbol type)
+                => type.TypeKind is TypeKind.Dynamic or TypeKind.Error;
+
+            void checkOperatorKinds()
+            {
+                // Dynamic built-in operators allow virtually all operations with all types.  So we do no further checking here.
+                if (isDynamicOrError(csharpReturnType) ||
```

Comment:
>isDynamicOrError(csharpReturnType) [](http://example.com/codeflow?start=20&length=34)

I think error types with known special type should go through additional validation. Effectively we know what they are.